benchmarks/src/main/java/io/druid/benchmark/FilteredAggregatorBenchmark.java

```diff
@@ -0,0 +1,317 @@
+            new RegexDimFilter("dimSequential", "X", null),
+            new SearchQueryDimFilter("dimSequential", new ContainsSearchQuerySpec("X", false), null),
+            new InDimFilter("dimSequential", Arrays.asList("X"), null)
+        )
+    );
+    filteredMetrics = new AggregatorFactory[1];
+    filteredMetrics[0] = new FilteredAggregatorFactory(new CountAggregatorFactory("rows"), filter);
+    incIndexFilteredAgg = makeIncIndex(filteredMetrics);
+
+    inputRows = new ArrayList<>();
+    for (int j = 0; j < rowsPerSegment; j++) {
+      InputRow row = gen.nextRow();
+      if (j % 10000 == 0) {
+        log.info(j + " rows generated.");
+      }
+      incIndex.add(row);
+      inputRows.add(row);
+    }
+
+    File tmpFile = Files.createTempDir();
+    log.info("Using temp dir: " + tmpFile.getAbsolutePath());
+    tmpFile.deleteOnExit();
+
+    indexFile = INDEX_MERGER_V9.persist(
+        incIndex,
+        tmpFile,
+        new IndexSpec()
+    );
+    qIndex = INDEX_IO.loadIndex(indexFile);
+
+    factory = new TimeseriesQueryRunnerFactory(
+        new TimeseriesQueryQueryToolChest(
+            QueryBenchmarkUtil.NoopIntervalChunkingQueryRunnerDecorator()
+        ),
+        new TimeseriesQueryEngine(),
+        QueryBenchmarkUtil.NOOP_QUERYWATCHER
+    );
+
+    BenchmarkSchemaInfo basicSchema = BenchmarkSchemas.SCHEMA_MAP.get("basic");
+    QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(Arrays.asList(basicSchema.getDataInterval()));
+    List<AggregatorFactory> queryAggs = new ArrayList<>();
+    queryAggs.add(filteredMetrics[0]);
+
+    query = Druids.newTimeseriesQueryBuilder()
+                  .dataSource("blah")
+                  .granularity(QueryGranularities.ALL)
+                  .intervals(intervalSpec)
+                  .aggregators(queryAggs)
+                  .descending(false)
+                  .build();
+  }
+
+  private IncrementalIndex makeIncIndex(AggregatorFactory[] metrics)
+  {
+    return new OnheapIncrementalIndex(
+        new IncrementalIndexSchema.Builder()
+            .withQueryGranularity(QueryGranularities.NONE)
+            .withMetrics(metrics)
+            .withDimensionsSpec(new DimensionsSpec(null, null, null))
+            .build(),
+        true,
+        false,
+        true,
+        rowsPerSegment
+    );
+  }
+
+  private static <T> List<T> runQuery(QueryRunnerFactory factory, QueryRunner runner, Query<T> query)
+  {
+    QueryToolChest toolChest = factory.getToolchest();
+    QueryRunner<T> theRunner = new FinalizeResultsQueryRunner<>(
+        toolChest.mergeResults(toolChest.preMergeQueryDecoration(runner)),
+        toolChest
+    );
+
+    Sequence<T> queryResult = theRunner.run(query, Maps.<String, Object>newHashMap());
+    return Sequences.toList(queryResult, Lists.<T>newArrayList());
+  }
+
+  // Filtered agg doesn't work with ingestion, cardinality is not supported in incremental index
+  /*
+  @Benchmark
```

Comment:
Also link to the github issue that describes what is broken here. If it does not exist then please file one :)